src/routes/rooms/update-room-by-id.ts

```diff
@@ -0,0 +1,67 @@
+import { Prisma } from '@prisma/client';
+import { Response } from 'express';
+import { prisma } from '../../prisma';
+import { RequestWithUser } from '../../types/request-with-user';
+import { paramIdValidationSchema } from './param-id.validation';
+import { roomNameValidatorSchema } from './room.validation';
+
+export const updateRoomNameById = async (
+  req: RequestWithUser,
+  res: Response
+) => {
+  const id = paramIdValidationSchema.safeParse(req.params?.id);
+  const body = roomNameValidatorSchema.safeParse(req.body);
+  const userId = req.user?.id;
+
+  if (!id.success) {
+    return res.status(400).json({ success: false, message: 'Invalid param' });
+  }
+
+  if (!body.success) {
+    return res
+      .status(400)
+      .json({ success: false, message: 'Invalid room request body' });
+  }
+
+  const room = await prisma.room.findUnique({
+    where: {
+      id: id.data,
+    },
+    select: {
+      id: true,
+      creatorId: true,
+    },
+  });
+
+  if (room === null) {
```

Comment:
Aqui daria pra usar [Boolean()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion) pra fazer o check
```suggestion
  if (!Boolean(room)) {
```

Reply:
Cero, vou corrigir!